src/main/java/org/broadinstitute/hellbender/utils/fermi/Assembly.java

```diff
@@ -0,0 +1,92 @@
+import java.util.HashMap;
+import java.util.List;
+
+/** an assembly is just a collection of contigs */
+public final class Assembly {
+    private final List<Contig> contigs;
+
+    Assembly( final List<Contig> contigs ) { this.contigs = contigs; }
+
+    public int getNContigs() { return contigs.size(); }
+    public Contig getContig( final int idx ) { return contigs.get(idx); }
+    public List<Contig> getContigs() { return Collections.unmodifiableList(contigs); }
+
+    /** a sequence of bases, coverage data, and connections to other contigs */
+    public final static class Contig {
+        private final byte[] sequence;
+        private final int nSupportingReads;
+        private final byte[] perBaseCoverage;
+        private List<Connection> connections;
+
+        Contig( final byte[] sequence, final int nSupportingReads, final byte[] perBaseCoverage ) {
+            this.sequence = sequence;
+            this.nSupportingReads = nSupportingReads;
+            this.perBaseCoverage = perBaseCoverage;
+        }
+
+        public byte[] getSequence() { return sequence; }
+        public int getNSupportingReads() { return nSupportingReads; }
+        public byte[] getPerBaseCoverage() { return perBaseCoverage; }
+        public List<Connection> getConnections() { return Collections.unmodifiableList(connections); }
+
+        void setConnections( final List<Connection> connections ) {
+            this.connections = connections;
+        }
+    }
+
+    /** a connection between contigs */
+    public final static class Connection {
+        private final Contig target;      // contig that overlaps the one that possesses this connection
+        private final int overlapLen;     // bases in common -- negative overlap lengths are legal, and represent gaps
+        private final boolean isRC;       // if target is a predecessor
+        private final boolean isTargetRC; // if connection is to RC of target contig
+
+        Connection( final Contig target, final int overlapLen, final boolean isRC, final boolean isTargetRC ) {
+            this.target = target;
+            this.overlapLen = overlapLen;
+            this.isRC = isRC;
+            this.isTargetRC = isTargetRC;
+        }
+
+        public Contig getTarget() { return target; }
+        public int getOverlapLen() { return overlapLen; }
+        public boolean isRC() { return isRC; }
+        public boolean isTargetRC() { return isTargetRC; }
+    }
+
+    public void writeGFA( final String fileName, final PipelineOptions pipelineOptions ) {
+        final HashMap<Contig, Integer> idMap = new HashMap<>((int)((contigs.size()*4L)/3) + 1);
+        int id = 0;
+        for (final Contig contig : contigs) {
+            idMap.put(contig, ++id);
+        }
+        try ( final Writer writer =
+                      new OutputStreamWriter(new BufferedOutputStream(BucketUtils.createFile(fileName, pipelineOptions))) ) {
+            writer.write("H\tVN:Z:1\n");
```

Comment:
tiny detail: the GFA spec seem to suggest version number as 1.0 with their example at the bottom.